crypto/ec/ec_asn1.c

```diff
@@ -1007,6 +1007,15 @@ EC_KEY *d2i_ECPrivateKey(EC_KEY **a, const unsigned char **in, long len)
     if (priv_key->parameters) {
         EC_GROUP_free(ret->group);
         ret->group = EC_GROUP_new_from_ecpkparameters(priv_key->parameters);
+        /*
+         * We depend on the ASN.1 flag being based on the actual encoding when
+         * the parameters are decoded.
+         * It can be of course overridden later for writing.
+         */
+        if (ret->group != NULL
+            && (EC_GROUP_get_asn1_flag(ret->group)
+                & OPENSSL_EC_NAMED_CURVE) == 0)
+            EC_KEY_set_flags(ret, EC_FLAG_EXPLICIT_CURVE_USED);
```

Comment:
I am afraid of adding further fields to the asn1 flag in EC_GROUP. Even the `_get/set_asn1_flag` name somewhat indicates that there is only a single flag and not multiple bit flags. I could add another internal flag member to the EC_GROUP structure if you do not like the implicit dependency on the asn1 flag being set in the new_from_ecpkparameters call.

Reply:
@romen would you be OK with my proposal?

Reply:
Yes, I think ideally we want the new flag to be read-only and initialized as close as possible to the point in which the parameters are parsed.

It makes absolutely sense to have a getter at the EC_KEY level, but I would prefer if the initialization of the truth value was done based on the parsed encoding rather than relying on the value of the output flag.

@t8m does that seem agreeable? I want the decision to be a consensus, not just a byproduct of my grumpiness 😜

Reply:
@romen Sure. I think your grumpiness is correct :laughing: 

What about this latest revision?